packages/context/test/unit/resolver.test.ts

```diff
@@ -285,6 +459,57 @@ describe('async constructor & sync property injection', () => {
   });
 });
 
+describe('async constructor injection with errors', () => {
+  let ctx: Context;
+
+  before(function() {
+    ctx = new Context();
+    ctx.bind('foo').toDynamicValue(
+      () =>
+        new Promise((resolve, reject) => {
+          setImmediate(() => {
+            reject(new Error('foo: error'));
+          });
+        }),
+    );
+  });
+
+  it('resolves properties and constructor arguments', async () => {
+    class TestClass {
+      constructor(@inject('foo') public foo: string) {}
+    }
+
+    try {
+      await instantiateClass(TestClass, ctx);
+    } catch (e) {
+      expect(e.message).to.eql('foo: error');
```

Comment:
Change this try-catch to be an expect.throw instead.

Reply:
Have you considered using [`Assertion#rejectedWith()`](https://shouldjs.github.io/#assertion-rejectedwith`) instead? I think something like this should work:

```ts
await expect(instantiateClass(TestClass, ctx))
  .to.be.rejectedWith('foo: error');
```